[VCDA-603] Catalog sharing fix for system tests,

rocknes opened the pull request:
* Fixed catalog sharing issue in system tests. Now users other than catalog author can access the test catalog.
* Tests in idisk_tests.py and vapp_tests.py now now run in VAPP_AUTHOR context.
* System Admin client is no longer a singleton. Each call to get_sys_admin_client() will now generate a new client. Added fallout code to logout the aforementioned client in various test suites.
* Fixed documentation in system tests.
* Fixed flake8 errors in search, client and cleanup tests.

Ran all the tests
(pyvcloud) C:\code\pyvcloud\system_tests>python -m unittest idisk_tests.py client_tests.py vapp_tests.py catalog_tests.py cleanup_test.py -v
All of them succeeded.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/pyvcloud/257)
<!-- Reviewable:end -->

hodgesrm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (After fixing noted items)

---

Reviewed 8 of 8 files at r1.
Review status: all files reviewed, 5 unresolved discussions (waiting on @rocknes and @ckolovson)

---

*[a discussion](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFPbWh6ET3KsukfnM3B:-LFPbWh77QG4uu0L_jHu:b-z6sg1d) (no related file):*
Can you add vapp_tests.py to the stable list in run_system_tests.sh and ensure it works?  This will cause your tests to build automatically.  Thanks!

---

*[pyvcloud/system_test_framework/environment.py, line 522 at r1](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFQRQu-6hTa7NtBJHzC:-LFQRQu-6hTa7NtBJHzD:b-gelsv9) ([raw file](https://github.com/vmware/pyvcloud/blob/6d03258e77c463b701a307a25527ffa6bd88c755/pyvcloud/system_test_framework/environment.py#L522)):*
> ```Python
>                     return
>             raise EntityNotFoundException('Catalog ' + catalog_name +
>                                           'doesn\'t exists.')
> ```

Nit: "exists" should be "exist"

---

*[system_tests/idisk_tests.py, line 134 at r1](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFQjauO6hBdnKck6LSY:-LFQjauO6hBdnKck6LSZ:b-gcx9eb) ([raw file](https://github.com/vmware/pyvcloud/blob/6d03258e77c463b701a307a25527ffa6bd88c755/system_tests/idisk_tests.py#L134)):*
> ```Python
> 
>         This test passes if the disk returned by the method is nor None, and
>         it's name matches the expected name of the disk.
> ```

it's => its

---

*[system_tests/idisk_tests.py, line 149 at r1](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFQjfIy60Fgmi-ASRnU:-LFQjfIy60Fgmi-ASRnV:b-gcx9eb) ([raw file](https://github.com/vmware/pyvcloud/blob/6d03258e77c463b701a307a25527ffa6bd88c755/system_tests/idisk_tests.py#L149)):*
> ```Python
> 
>         This test passes if the disk returned by the method is nor None, and
>         it's id matches the expected id of the disk.
> ```

it's => its

---

*[system_tests/idisk_tests.py, line 278 at r1](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFQjkhu3xX2xf1fTYXQ:-LFQjkhu3xX2xf1fTYXR:b-gcx9eb) ([raw file](https://github.com/vmware/pyvcloud/blob/6d03258e77c463b701a307a25527ffa6bd88c755/system_tests/idisk_tests.py#L278)):*
> ```Python
>         """Test the  method vapp.update_disk().
> 
>         Invoke the method for the first independent disk, to update it's name,
> ```

it's => its

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/257#-:-LFQk7K8C3rF27n0Nri-:byfj9l9)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Review status: 6 of 9 files reviewed, 5 unresolved discussions (waiting on @hodgesrm, @rocknes, and @ckolovson)

---

*[a discussion](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFPbWh6ET3KsukfnM3B:-LFTFRiN2-vn7GfVLklp:b-rxk4t8) (no related file):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

Can you add vapp_tests.py to the stable list in run_system_tests.sh and ensure it works?  This will cause your tests to build automatically.  Thanks!
</blockquote></details>

vApp tests are already part of stable tests. Catalog tests were missing, added them.

---

*[pyvcloud/system_test_framework/environment.py, line 522 at r1](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFQRQu-6hTa7NtBJHzC:-LFTEqkN1Zt4gJoiA-7c:bt2vr1b) ([raw file](https://github.com/vmware/pyvcloud/blob/6d03258e77c463b701a307a25527ffa6bd88c755/pyvcloud/system_test_framework/environment.py#L522)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

Nit: "exists" should be "exist"
</blockquote></details>

Fixed.

---

*[system_tests/idisk_tests.py, line 134 at r1](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFQjauO6hBdnKck6LSY:-LFTEYw04eawsjjzHY8b:b-pi05tv) ([raw file](https://github.com/vmware/pyvcloud/blob/6d03258e77c463b701a307a25527ffa6bd88c755/system_tests/idisk_tests.py#L134)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

it's => its
</blockquote></details>

It seems I make this specific grammatical mistake too often. I should stick a sticky note on my monitor :P

Fixed.

---

*[system_tests/idisk_tests.py, line 149 at r1](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFQjfIy60Fgmi-ASRnU:-LFTF3pS32JXVRlc6uN8:bt2vr1b) ([raw file](https://github.com/vmware/pyvcloud/blob/6d03258e77c463b701a307a25527ffa6bd88c755/system_tests/idisk_tests.py#L149)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

it's => its
</blockquote></details>

Fixed.

---

*[system_tests/idisk_tests.py, line 278 at r1](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFQjkhu3xX2xf1fTYXQ:-LFTF2WH6ZiR5epbhi3h:bt2vr1b) ([raw file](https://github.com/vmware/pyvcloud/blob/6d03258e77c463b701a307a25527ffa6bd88c755/system_tests/idisk_tests.py#L278)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

it's => its
</blockquote></details>

Fixed.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/257)*
<!-- Sent from Reviewable.io -->

hodgesrm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 6 of 9 files reviewed, all discussions resolved (waiting on @hodgesrm and @ckolovson)

---

*[a discussion](https://reviewable.io/reviews/vmware/pyvcloud/257#-LFPbWh6ET3KsukfnM3B:-LFTGmeE25dfWLsII9_L:bck1xlf) (no related file):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

vApp tests are already part of stable tests. Catalog tests were missing, added them.
</blockquote></details>

Eek.  Thanks for correcting my review.  :)

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/pyvcloud/257#-:-LFTGutY3f174JDBUTZ6:bnfp4nl)*
<!-- Sent from Reviewable.io -->
